rename to CLOSE_STREAM, make error code optional (#11)
quicwg/reliable-stream-reset: draft-ietf-quic-reliable-stream-reset.md

the server, the server MUST not disable this extension on the resumed
connection.

# CLOSE_STREAM Frame

Conceptually, the CLOSE_STREAM frame is a RESET_STREAM frame with an
added Reliable Size field.

~~~
CLOSE_STREAM Frame {
  Type (i) = 0x20..0x21,
  Stream ID (i),

>   [Application Protocol Error Code (i)],

Reviewer: I would prefer if this were unconditional.  A value of 0 (or whatever application error equates to no error) is fine.

I see why you made this choice, but if the goal is to send STREAM+FIN, then STREAM+FIN exists.  This frame only makes sense if reliable size < final size.

If the Reliable Size is larger than the Final Size, the receiver MUST close the
connection with a connection error of type FRAME_ENCODING_ERROR.

CLOSE_STREAM frames are ack-eliciting. When lost, they MUST be retransmitted,

> unless another frame that causes a state transition to the Data Sent or Reset
> Sent State was sent for the same stream (see {{multiple-frames}}).

Reviewer 2: How about talking about acknowledgement and terminal states?

```suggestion
unless the stream state has transitioned to "Data Recvd" or "Reset Recvd" due
to transmission and acknowledgement of other frames (see {{multiple-frames}}).
```

IMO, current text is a bit ambiguous in situations like when you send STREAM + FIN, then CLOSE_STREAM, after which you receive an ACK for STREAM + FIN. In such case, acknowledgment of STREAM + FIN changes the stream state to "Data Recvd" at which point the sender learns that retransmission of CLOSE_STREAM is unnecessary.

# Closing Streams

## Without an Error

> When closing a stream with an error, the node has the choise between a STREAM

Reviewer: choice

frame that carries the FIN bit and a CLOSE_STREAM frame of type 0x20.

The CLOSE_STREAM frame can be used to reduce the reliable offset after a
STREAM frame with a FIN bit has been sent. If STREAM frames containing data
up to that byte offset are lost, the initiator MUST retransmit this data, as

> described in (Section 13.3 of {{!RFC9000}}). Data sent beyond that byte offset

Reviewer: `{{Section 13.3 of !RFC9000}}`

SHOULD NOT be retransmitted.

## Using an Error Code

When resetting a stream, the node has the choice between using a RESET_STREAM
frame and a CLOSE_STREAM frame of type 0x21. When using a RESET_STREAM frame,

> the behavior is unchanged from the behavior described in ({{!RFC9000}}).

Reviewer: Lose the parentheses.

> The initiator MUST guarantee reliable delivery of stream data of at least

Reviewer 2: ```suggestion
When using the CLOSE_STREAM frame, the initiator MUST guarantee reliable delivery of stream data of at least
```
